tests/utilities/test_cli.py

```diff
@@ -673,6 +699,7 @@ def __init__(self, optim1: dict, optim2: dict, scheduler: dict):
             self.scheduler = instantiate_class(self.optim1, scheduler)
 
     cli_args = [
+        "fit",
```

Comment:
@mauvilsa This test is failing now (`test_lightning_cli_optimizers_and_lr_scheduler_with_link_to`):

```python
test_cli.py:711: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../pytorch_lightning/utilities/cli.py:293: in __init__
    self.instantiate_classes()
../../pytorch_lightning/utilities/cli.py:401: in instantiate_classes
    self.config_init = self.parser.instantiate_classes(self.config)
../../../../.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/core.py:1123: in instantiate_classes
    cfg[subcommand] = subparser.instantiate_classes(cfg[subcommand], instantiate_groups=instantiate_groups)
../../../../.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/core.py:1117: in instantiate_classes
    component.instantiate_class(component, cfg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

group = <jsonargparse.core._ArgumentGroup object at 0x1385439d0>
cfg = {'config': None, 'lr_scheduler': {'gamma': 0.2, 'last_epoch': -1, 'verbose': False}, 'optim1': {'amsgrad': False, 'bet...class_path': 'torch.optim.SGD', 'init_args': {'dampening': 0, 'lr': 0.01, 'momentum': 0, 'nesterov': False, ...}}, ...}

    def group_instantiate_class(group, cfg):
        try:
            value, parent, key = _get_key_value(cfg, group.dest, parent=True)
        except KeyError:
            value = {}
            parent = cfg
            key = group.dest
            assert '.' not in key
>       parent[key] = group.group_class(**value)
E       TypeError: __init__() missing 3 required positional arguments: 'optim1', 'optim2', and 'scheduler'

../../../../.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/signatures.py:550: TypeError
```

Does `link_to` not work properly with subcommand parsers or is there something wrong with this PR?

Reply:
Could be that argument linking does not work well with the subcommands. Need to look into it.

Reply:
Opened https://github.com/omni-us/jsonargparse/issues/82 to track it